File: src/main/java/com/pspl/expense_voucher_system/service/DashboardService.java
package com.pspl.expense_voucher_system.service;

import com.pspl.expense_voucher_system.dto.AccountsDashboardResponse;
import com.pspl.expense_voucher_system.dto.DirectorDashboardResponse;
import com.pspl.expense_voucher_system.dto.EmployeeDashboardResponse;

/**
 * DashboardService defines summary views for each role-specific dashboard.
 */
public interface DashboardService {

	EmployeeDashboardResponse getEmployeeDashboard();

	DirectorDashboardResponse getDirectorDashboard();

	AccountsDashboardResponse getAccountsDashboard();
}

File: frontend/src/services/dashboardService.js
import apiClient from "@/api/axios";

/**
 * dashboardService fetches the role-specific dashboard summaries from the backend.
 */
export async function getEmployeeDashboard() {
  const response = await apiClient.get("/dashboard/employee");
  return response.data;
}

/**
 * Fetches the director dashboard summary.
 */
export async function getDirectorDashboard() {
  const response = await apiClient.get("/dashboard/director");
  return response.data;
}

/**
 * Fetches the accounts dashboard summary.
 */
export async function getAccountsDashboard() {
  const response = await apiClient.get("/dashboard/accounts");
  return response.data;
}
